velox/exec/tests/TableWriterTest.cpp

```diff
@@ -1553,6 +1597,37 @@ TEST_P(UnpartitionedTableWriterTest, immutableSettings) {
   }
 }
 
+TEST_P(BucketedUnpartitionedTableWriterTest, bucketNonPartitioned) {
+  SCOPED_TRACE(testParam_.toString());
+  auto input = makeVectors(1, 100);
+  createDuckDbTable(input);
+
+  auto outputDirectory = TempDirectoryPath::create();
+  setBucketProperty(
+      bucketProperty_->kind(),
+      bucketProperty_->bucketCount(),
+      bucketProperty_->bucketedBy(),
+      bucketProperty_->bucketedTypes(),
+      bucketProperty_->sortedBy());
+  auto plan = createInsertPlan(
+      PlanBuilder().values({input}),
+      rowType_,
+      outputDirectory->getPath(),
+      {},
+      bucketProperty_,
+      compressionKind_,
+      getNumWriters(),
+      connector::hive::LocationHandle::TableType::kExisting,
```

Comment:
https://github.com/facebookincubator/velox/pull/9740 only test the `connector::hive::LocationHandle::TableType::kNew` use case.